docs/eloquent-models/model-class.txt

```diff
@@ -333,3 +334,94 @@ models that the prune action deletes:
    :emphasize-lines: 5,10,12
    :dedent:
 
+.. _laravel-schema-versioning:
+
+Create a Versioned Model Schema
+-------------------------------
+
+You can implement a schema versioning pattern into your application by
+using the ``HasSchemaVersion`` trait on an Eloquent model. You might
+choose to implement a schema version to organize or standardize a
+collection that contains data with different schemas.
+
+.. tip::
+
+   To learn more about schema versioning, see the :manual:`Model Data for
+   Schema Versioning </tutorial/model-data-for-schema-versioning/>`
+   tutorial in the {+server-docs-name+}.
+
+To use this feature with models that use MongoDB as a database, add the
+``MongoDB\Laravel\Eloquent\HasSchemaVersion`` import to your model.
+Then, set the ``SCHEMA_VERSION`` constant to ``1`` to set the first
+schema version on your collection. If your collection evolves to contain
+multiple schemas, you can update the value of the ``SCHEMA_VERSION``
+constant in subsequent model classes.
+
+When creating your model, you can define the ``migrateSchema()`` method
+to specify a migration to the current schema version upon saving a
+model. In this method, you can specify the changes to make to an older
+model to update it to match the current schema version.
+
+Schema Versioning Example
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In this sample situation, you are working with a collection that was
+first modeled by the following class:
+
+.. literalinclude:: /includes/eloquent-models/PlanetSchemaVersion1.php
+   :language: php
+   :dedent:
+
+Now, you want to implement a new schema version on the collection.
+You can define the new model class with the following behavior:
+
+- Implements the ``HasSchemaVersion`` trait and sets the current
+  ``SCHEMA_VERSION`` to ``2``
+
+- Defines the ``migrateSchema()`` method to migrate models in which the
+  schema version is less than ``2`` to have a ``galaxy`` field that has a value
+  of ``'Milky Way'``
+
+.. literalinclude:: /includes/eloquent-models/PlanetSchemaVersion2.php
+   :language: php
+   :emphasize-lines: 10,12,20
+   :dedent:
+
+When you save or retrieve a model that does not have a schema version
+specified, Laravel assumes that it follows the latest schema version.
```

Comment:
We automatically save the current "SCHEMA_VERSION". When a model is retrieved, if it doesn't contain the schema_version property, we assume it was inserted before the model was versioned, so we assume its version `0` and run the full migration.

```suggestion
When you save a model that does not have a schema version specified,
Laravel assumes that it follows the latest schema version.
```

Reply:
Will update the copy